addons/resource/models/resource.py

```diff
@@ -101,7 +97,7 @@ def interval_remove_leaves(self, cr, uid, interval, leave_intervals, context=Non
         if leave_intervals is None:
             leave_intervals = []
         intervals = []
-        leave_intervals = self.interval_clean(leave_intervals)
+        leave_intervals = self._model.interval_clean(leave_intervals)
```

Comment:
Interval clean is an utility method. I don't think that you need to select the model. Only the intervals field is taken into account